src/main/scala/sigmastate/lang/SigmaBinder.scala

```diff
@@ -129,6 +129,20 @@ class SigmaBinder(env: Map[String, Any], builder: SigmaBuilder) {
         Some(newBlock)
       else
         None
+    case e @ Apply(ApplyTypes(DeserializeSym, targs), args) =>
+      if (targs.length != 1)
+        error(s"Wrong number of type arguments in $e: expected one type argument")
+      if (args.length != 1)
+        error(s"Wrong number of arguments in $e: expected one argument")
+      val str = args.head match {
+        case StringConstant(s) => s
+        case _ =>
+          error(s"invalid argument in $e: expected a string constant")
+      }
+      val bytes = Base58.decode(str).get
+      // todo handle decoding failure
+      Some(
+        ValueSerializer.deserialize(bytes))
```

Comment:
@ergomorphic Is that what you had in mind for `deserialize`? Or should it be done by the interpreter?

Reply:
@greenhat yes, you are right. However, this will be currently a little ad-hoc. In C++ it would be constexpr function, but we don't have such modifier.